### ISSUE TITLE
Translations & more info

### ISSUE THREAD

**Issue by Inrego** [mmcc-xx/BirdCAGE#20]
So.. after setting this up, I've realized that I have no idea what all the birds in my area are called in English. So I'm checking each and every one of them in wikipedia and then change it to the page of my country.
Would it be possible to show the localized names of the birds in the UI? I'm not completely sure about any online resources that would be reliable to implement while free to use.
Also as a side-note. It would be great to be able to read more about the bird and see a photo without having to google it. Again, it would probably require an external resource. Or at the very least, maybe a link to the bird on wikipedia (or another online bird database, whichever you find suitable).

**Comment by ministryofsillywalks** [comment]
Maybe the birdweather API can be used for more info:
https://app.birdweather.com/api/v1/index.html#species-species-get
Seems to include a picture and link to the wiki page.

**Comment by Inrego** [comment]
Great! For the translation part, I looked around and there doesn't seem to be anything in the birdweather api to get localized names.
But since it has the Wikipedia url, I suppose that could be used in combination with the MediaWiki API to get the localized name:
https://www.mediawiki.org/wiki/API:Langlinks

Example:
https://en.wikipedia.org/w/api.php?action=query&titles=Common%20chiffchaff&prop=langlinks&lllimit=500

**Comment by Inrego** [comment]
And in order to take it easy on those api endpoints, I guess a database table could be used to store the localized names, so we only request the api's when we encounter new species or change the language. So with a primary key of scientific name + language code and then two more columns for localized name and localized wiki url. And then probably remove common name from detections table and just look up the name in this new table whenever the common name is to be displayed.

This way, it would also work on historic detections (detections made before this localization update eventually is released)

**Comment by Inrego** [comment]
On second thought, maybe keep the common name provided by the local birdnetserver container, so that it's still keeping the same functionality for people who want completely local operation without accessing external services.

**Comment by mmcc-xx** [comment]
The BirdNET Analyzer provides localized labels, but I don't know how to ask it to use them :) I'll look into it.

**Comment by ministryofsillywalks** [comment]
@mmcc-xx I saw you implemented locales, however setting my locale to de didn't result in any German names in the detection. All stays English.

**Comment by ministryofsillywalks** [comment]
these are the logs from the birdnetserver with german locale

```

####################  2023-05-27 07:25:33.833550  ####################
{'lat': 'XXXX', 'lon': 'XXXX', 'week': 21, 'overlap': '0', 'sensitivity': '1', 'sf_thresh': '0.03', 'pmode': 'max', 'num_results': 5, 'locale': 'de', 'save': False}
Analyzing /tmp/tmpevvlc29_.wav
Finished /tmp/tmpevvlc29_.wav in 0.48 seconds
Results:
{}
####################  2023-05-27 07:25:46.395096  ####################
{'lat': 'XXXX', 'lon': 'XXXX', 'week': 21, 'overlap': '0', 'sensitivity': '1', 'sf_thresh': '0.03', 'pmode': 'max', 'num_results': 5, 'locale': 'de', 'save': False}
Analyzing /tmp/tmp3bwqpasp.wav
Finished /tmp/tmp3bwqpasp.wav in 0.47 seconds
Results:
{}
```

**Comment by ministryofsillywalks** [comment]
siwthing back to locale "en" and its working again:

```
Not Getting Translated Labels
####################  2023-05-27 07:29:02.173903  ####################
{'lat': 'XXXX', 'lon': 'XXXX', 'week': 21, 'overlap': '0', 'sensitivity': '1', 'sf_thresh': '0.03', 'pmode': 'max', 'num_results': 5, 'locale': 'en', 'save': False}
Analyzing /tmp/tmp3hssx9tv.wav
Finished /tmp/tmp3hssx9tv.wav in 9.27 seconds
Results:
{'0.0;3.0': [('Passer domesticus_House Sparrow', 0.3679), ('Passer montanus_Eurasian Tree Sparrow', 0.0711)], '12.0;15.0': [('Turdus merula_Eurasian Blackbird', 0.1636)], '6.0;9.0': [('Phoenicurus ochruros_Black Redstart', 0.0439), ('Delichon urbicum_Common House-Martin', 0.0379)]}
####################  2023-05-27 07:29:12.538515  ####################
{'lat': 'XXXX', 'lon': 'XXXX', 'week': 21, 'overlap': '0', 'sensitivity': '1', 'sf_thresh': '0.03', 'pmode': 'max', 'num_results': 5, 'locale': 'en', 'save': False}
Analyzing /tmp/tmpzh2c0sff.wav
Finished /tmp/tmpzh2c0sff.wav in 0.55 seconds
Results:
{'0.0;3.0': [('Passer domesticus_House Sparrow', 0.2082), ('Turdus merula_Eurasian Blackbird', 0.1698)], '12.0;15.0': [('Oriolus oriolus_Eurasian Golden Oriole', 0.0498), ('Delichon urbicum_Common House-Martin', 0.0328)], '9.0;12.0': [('Hippolais icterina_Icterine Warbler', 0.0111)]}
```

**Comment by mmcc-xx** [comment]
The law of unintended consequences bit me. I had made a change on the BirdNET-Analyzer side so the results would come out localized, and that worked so well that I made another change on the BirdNET-Analyzer side so the birds of the week list would come out localized, which also worked. However, it turned out that Change 2 broke Change 1.

I pushed a new birdnetserver image that undoes change 2. Please confirm that you are seeing localized results and I'll leave this open until I get the birds of the week list working correctly too.

**Comment by ministryofsillywalks** [comment]
Working now! Well done! Now I have english and german detections in my list.
Will deleting the db start from scratch, with just german detections?

**Comment by mmcc-xx** [comment]
Yes but you'll have to redo your preferences and stream definitions.

If you are so inclined you could shut down the app, access the database with the sqlite3 command line utility, and execute...
DELETE FROM detections;

Also also delete all of the mp3 files from the detections directory.

If I was doing this the right way, I supposed I'd just store the scientific name and look up the localized names on the fly. Hmm.

**Comment by Inrego** [comment]
> If I was doing this the right way, I supposed I'd just store the scientific name and look up the localized names on the fly. Hmm.

Exactly as I suggested :D
